File: Python/_118_Pascal's_Triangle/solution.py
class Solution:
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if numRows == 0:
            return []
        else:
            result = []
            firstlayer = [1]
            result.append(firstlayer)
            for _ in range(0, numRows - 1):
                firstlayer = self.singlelayer(firstlayer)
                result.append(firstlayer)
            return result
        
    def singlelayer(self, prelayer):
        nextlayer = []
        nextlayer.append(1)
        for i in range(0, len(prelayer) - 1):
            nextlayer.append(prelayer[i] + prelayer[i + 1])
        nextlayer.append(1)
        return nextlayer

File: Java/_118_Pascal's_Triangle/Solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        if(numRows == 0)
            return result;
        else
        {
            List<Integer> firstLayer = new ArrayList<Integer>();
            firstLayer.add(1);
            result.add(firstLayer);
            for(int i = 0; i < numRows - 1; i++)
            {
                firstLayer = singleLayer(firstLayer);
                result.add(firstLayer);
            }
            return result;
        }
    }
    
    public List<Integer> singleLayer(List<Integer> preLayer) {
        List<Integer> nextLayer = new ArrayList<Integer>();
        nextLayer.add(1);
        for(int i = 0; i < preLayer.size() - 1; i++)
            nextLayer.add(preLayer.get(i) + preLayer.get(i + 1));
        nextLayer.add(1);
        return nextLayer;
    }
}